src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -2375,14 +2369,21 @@ private unsafe void CreateUriInfo(Flags cF)
 
         Done:
             cF |= Flags.MinimalUriInfoSet;
-            info.DnsSafeHost = _dnsSafeHost;
-            lock (_string)
+
+            Debug.Assert(sizeof(Flags) == sizeof(ulong));
```

Comment:
Worth asserting here that we're not in a ctor?  That might help to highlight why the subsequent synchronization is being used.  Or is it possible we are being called from a ctor here?

Reply:
It's possible this is called from the constructor - if the uri contains any non-ascii characters a full parse will be triggered at ctor time